titus-client/src/main/java/com/netflix/titus/runtime/endpoint/v3/grpc/GrpcJobManagementModelConverters.java

```diff
@@ -388,20 +389,27 @@ private static BasicContainer toCoreBasicContainer(com.netflix.titus.grpc.protog
     }
 
     private static Volume toCoreVolume(com.netflix.titus.grpc.protogen.Volume grpcVolume) {
-        return Volume.newBuilder()
-                .withName(grpcVolume.getName())
-                .withSharedContainerVolumeSource(toCoreSharedVolumeSource(grpcVolume.getSharedContainerVolumeSource()))
-                .build();
+        if (grpcVolume.getSharedContainerVolumeSource() != null) {
+            return SharedContainerVolume.newBuilder()
+                    .withName(grpcVolume.getName())
+                    .withSharedContainerVolumeSource(toCoreSharedVolumeSource(grpcVolume.getSharedContainerVolumeSource()))
+                    .build();
+        } else {
+            // Currently the shared volume source is the only kind of volume we support
+            return null;
```

Comment:
This is dangerous. We should either do invariant check and throw exception or return `Optional<Volume>`

Reply:
Let me figure out how to do one of those. I agree I don't like having the null here, but on the other hand it doesn't sound too crazy to have the control plane tolerate (but just not understand) new volumes?